src/controllers/apps/chat-app/message.controllers.js

```diff
@@ -154,5 +158,89 @@ const sendMessage = asyncHandler(async (req, res) => {
     .status(201)
     .json(new ApiResponse(201, receivedMessage, "Message saved successfully"));
 });
+const deleteMessage = asyncHandler(async (req, res) => {
+  //controller to delete chat messages and attachments
 
-export { getAllMessages, sendMessage };
+  const { chatId, messageId } = req.params;
+
+  //Find the chat based on chatId
+
+  const chat = await Chat.findOne({
+    _id: new mongoose.Types.ObjectId(chatId),
+  });
+
+  if (!chat) {
+    throw new ApiError(404, "Chat does not exist");
+  }
+  //Find the message based on message id
+
+  const message = await ChatMessage.findOne({
+    _id: new mongoose.Types.ObjectId(messageId),
+  });
+
+  if (!message) {
+    throw new ApiError(404, "Message does not exist");
+  }
+
+  // Check if user is the sender of the message
+  if (message.sender.toString() !== req.user._id.toString()) {
+    throw new ApiError(
+      401,
+      "You are not authorised to delete the message ,you are not the sender"
+    );
+  }
+
+  if (
+    !chat.participants.some(
```

Comment:
don't check this separately add this check in `Chat` find query itself to avoid moving ahead if user is not part of participants array.